src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+###
+# Computing Groebner fans in Oscar
+# ================================
+#
+# For a detailed writeup of the algorithm please see Anders Jensen's PhD dissertation
+#   "Algorithmic Aspects of Groebner Fans and Tropical Varieties"
+#
+# In tropical terminology: Groebner fans contain tropicalizations in the constant
+#   coefficient case. The convention is always max, as in OSCAR leading monomials
+#   under weighted orderings are the ones with maximal weighted degree.
+#
+###
+
+# Computes the space of weight vectors w.r.t. which G is weighted homogeneous
+@doc raw"""
+    homogeneity_space(G::Vector{<:MPolyElem})
+
+A `Cone` that is the space of all weight vectors under which all elements of `G` are weighted homogeneous.  The cone will be a linear subspace without any rays.
+
+# Note
+If `G` is a reduced Groebner basis of an ideal `I` with respect a global ordering, then the returned cone is the space of all weight vectors with respect to which `I` is weighted homogeneous.
+
+# Examples
+```jldoctest
+julia> Qx,(x1,x2,x3) = PolynomialRing(QQ,["x1","x2","x3"])
+(Multivariate Polynomial Ring in x1, x2, x3 over Rational Field, QQMPolyRingElem[x1, x2, x3])
+
+julia> G1 = [x1+x2,x2+x3,x3]
+3-element Vector{QQMPolyRingElem}:
+ x1 + x2
+ x2 + x3
+ x3
+
+julia> rays_modulo_lineality(homogeneity_space(G1))
+(rays_modulo_lineality = RayVector{QQFieldElem}[], lineality_basis = RayVector{QQFieldElem}[[1, 1, 1]])
+
+julia> G2 = [x1^2+x2,x2^2+x3,x3]
+3-element Vector{QQMPolyRingElem}:
+ x1^2 + x2
+ x2^2 + x3
+ x3
+
+julia> rays_modulo_lineality(homogeneity_space(G2))
+(rays_modulo_lineality = RayVector{QQFieldElem}[], lineality_basis = RayVector{QQFieldElem}[[1//4, 1//2, 1]])
+
+julia> G3 = [x1^2+x2,x2^2+x3,x3+1]
+3-element Vector{QQMPolyRingElem}:
+ x1^2 + x2
+ x2^2 + x3
+ x3 + 1
+
+julia> rays_modulo_lineality(homogeneity_space(G3))
+(rays_modulo_lineality = RayVector{QQFieldElem}[], lineality_basis = RayVector{QQFieldElem}[])
+
+```
+"""
+function homogeneity_space(G::Vector{<:MPolyElem})
+    inequalities = fmpq_mat(0,0)
+    equations = Vector{Vector{Int}}()
+    for g in G
+        alpha,tail_exponents = Iterators.peel(exponents(g))
+        for beta in tail_exponents
+            push!(equations,beta-alpha)
+        end
+    end
+    return cone_from_inequalities(inequalities,matrix(QQ,equations))
+end
+
+
+
+@doc raw"""
+    homogeneity_vector(G::Vector{<:MPolyElem})
+
+If it exists, a positive `Vector{fmpz}` under which every of `G` is weighted homogeneous.  Otherwise, `nothing`.
```

Comment:
Wouldn't it make more sense to return a zero vector instead of `nothing` if no such homgeneity vector exists?

Reply:
That is very true.  I guess I just wanted to play around with type dispatch and use it to distinguish between the (quasi-)homogeneous and the non-homogeneous case.